### PR TITLE
fix: handle when body is nil & add stop to goroutines

### PR DESCRIPTION
**Pull request by bryancua**

### CODE REVIEW

File: handler-host.go [hunk 1]
```diff
@@ -66,12 +66,19 @@ type HostHandler struct {
...
+	db                *sql.DB
 
 	// List of filenames/Ids that are currently being sent to the server,
 	// so they do not get sent again.
 	currentlySending   []string
 	currentlySendingMu sync.RWMutex
+
+	// Channel to stop processing
+	die           chan bool
+	stopHeartbeat chan bool
```
Comment (tmathews) [inline handler-host.go]: Do we need so many? Is it risky to reuse one?

**Review by tmathews — APPROVED** [review]

File: handler-host.go [hunk 2]
```diff
@@ -66,12 +66,19 @@ type HostHandler struct {
...
-	db *sql.DB
+	db                *sql.DB
 
 	// List of filenames/Ids that are currently being sent to the server,
 	// so they do not get sent again.
 	currentlySending   []string
 	currentlySendingMu sync.RWMutex
+
+	// Channel to stop processing
+	die           chan bool
```
Comment (lostlevels) [inline handler-host.go:77]: Why are there so many channels? Can just use one if the only intention is to stop things.

File: handler-host.go [hunk 3]
```diff
@@ -172,6 +179,21 @@ VALUES (?, ?, ?, ?)
...
 }
 
+func (h *HostHandler) initStopChannels() {
+	h.die = make(chan bool, 2)
+	h.stopHeartbeat = make(chan bool, 2)
+	h.stopSend = make(chan bool, 2)
+	h.stopRead = make(chan bool, 2)
+}
+
+func (h *HostHandler) Close() {
```
Comment (lostlevels) [inline handler-host.go:182]: I would expect a function called Close to close the channels.

**Review by lostlevels — CHANGES_REQUESTED** [review]

File: handler-host.go (hunk 3 above)
Comment (tmathews) [reply handler-host.go:182]: Agreed

File: handler-host.go (hunk 2 above)
Comment (bryancua) [reply handler-host.go:77]: do channels send to all subscribers?

Comment (bryancua) [reply handler-host.go:77]: i guess i can just pass in x values instead of passing a value to multiple channels...

File: handler-host.go (hunk 1 above)
Comment (bryancua) [reply handler-host.go]: bad habit.

**Review by lostlevels — APPROVED** [review]

**Pull request merged**